src/main/scala/com/imaginea/activegrid/core/models/Site1.scala

```diff
@@ -99,10 +102,18 @@ object Site1 {
       val mapPrimitives = Map("siteName" -> entity.siteName, "groupBy" -> entity.groupBy)
       val node = Neo4jRepository.saveEntity[Site1](label, entity.id, mapPrimitives)
       val relationship_inst = "HAS_Instance"
-      entity.instances.foreach { instance =>
-        val instanceNode = instance.toNeo4jGraph(instance)
-        Neo4jRepository.setGraphRelationship(node, instanceNode, relationship_inst)
+      val executorService = Executors.newFixedThreadPool(Constants.THREAD_POOL_SIZE)
```

Comment:
As I had previously thought it would invoke more number of threads as per the number of elements in the list,  I have used Executor service then. Now i have read that it will have thread pool, so i have changed to Futures.